paper2remarkable/utils.py

```diff
@@ -188,6 +192,63 @@ def upload_to_remarkable(filepath, remarkable_dir="/", rmapi_path="rmapi"):
     logger.info("Upload successful.")
 
 
+def list_usb_dir(guid):
+    url = REMARKABLE_USB_URL + "/documents/"
+    if guid:
+        url += guid + "/"
+    response = requests.post(url)
+    return response.json()
+
+
+def crawl_usb_dirs(remarkable_dir):
+    parts = [p for p in remarkable_dir.split("/") if p]
+    guid = None
+    while len(parts) > 0:
+        part = parts.pop(0)
+        data = list_usb_dir(guid)
+        guid = None
+        for item in data:
+            if item["VissibleName"] == part:
+                guid = item["ID"]
+        if not guid:
+            raise RemarkableError(
+                "Directory %s does not exist on reMarkable" % remarkable_dir
+            )
+    # necessary because the last listed dir is the one uploaded to
+    list_usb_dir(guid)
+
+
+def upload_to_remarkable_usb(filepath, remarkable_dir="/"):
+    # https://remarkable.guide/tech/usb-web-interface.html
+    logger.info("Starting upload to reMarkable")
+
+    # Check if the remarkable dir exists (no creation through usb api)
+    crawl_usb_dirs(remarkable_dir)
+
+    headers = {
```

Comment:
These headers are taken from https://remarkable.guide/tech/usb-web-interface.html#post-http-10-11-99-1-upload

I am not sure, why they are using these specific headers, but they work in practice.